lib/games/games_module.rb

```diff
@@ -17,19 +16,44 @@ def add_game
     print 'Please state how many years ago it was published: '
     published_date = gets.chomp
     new_game = Game.new(title, multiplayer, last_played_at, published_date)
-    @game_list.push(new_game)
+    @game_list.push(title: new_game.title, multiplayer: new_game.multiplayer, last_played_at: new_game.last_played_at, published_date: new_game.published_date)
     puts 'Game created succesfully'
     puts ''
   end
 
   def list_all_games
-    @game_list.each do |game|
-      puts "Title: \"#{game.title}\", Multiplayer: #{game.multiplayer}, Last Time Played:#{game.last_played_at} years ago, Published #{game.published_date} years ago"
+    @game_list = JSON.parse(File.read('./lib/games/json_data/games.json'))
+    if @game_list.empty?
+      puts 'There are no games in the list yet, please add one first'
+    else
+      @game_list.each do |game|
+        puts "Title: #{game['title']}, Multiplayer: #{game['multiplayer']}, Last Time Played:#{game['last_played_at']} years ago, Published #{game['published_date']} years ago"
+      end
     end
   end
```

Comment:
I would suggest changing line 25, it would be better to read a file content in a separated method. Just to follow the principle that a method should do only what the name says it does. 

There's also another issue with that line, when you have no files it gives you an Error. You could try to wrap it in a begin/rescue/else method to make sure the error is addressed when that file doesn't exist. 

![image](https://user-images.githubusercontent.com/85680436/144165033-12d964cd-c29e-45f9-b5a5-70b26d39c7b3.png)

Reply:
Great advice, I moved that line to read_game_data method 👍